python/google/protobuf/text_format.py

```diff
@@ -469,7 +469,7 @@ def Merge(text,
     ParseError: On text parsing problems.
   """
   return MergeLines(
-      text.split('\n'),
+      re.findall('(?:[^\"\n]|(?:\"(?:[^\"]|[\n])*\"))*\n?', text),
```

Comment:
I think we would want to make the same change in the Parse() method above for consistency.

Could you perhaps also add a comment that breaks down what that regex is doing?